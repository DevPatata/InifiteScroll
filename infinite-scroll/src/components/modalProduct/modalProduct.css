.background{
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  background-color: gray;
  opacity: 50%;
  z-index: 1;
}

.modal-position{
  z-index: 2;
  position: fixed;
  top: 30%;
}

.modal-container{
  background-color: #fff;
  padding: 1rem 2rem;
  border-radius: 5px;
  width: 50%;
}

.modal-close{
  display: none;
}

.div-cotainer{
  display: flex;
  justify-content: center;
}

.span-close{
  float: right;
  cursor: pointer;
}

.span-close:hover{
  color: var(--color-grey);
}



