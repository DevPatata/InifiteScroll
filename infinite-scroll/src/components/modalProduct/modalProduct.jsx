import React from 'react'
import './modalProduct.css'

export const ModalProduct = ({ children, isOpen = false, closeModal = () => { } }) => {
  return (
    <article data-testid='modal-test-id' className={`modal-close ${isOpen && 'div-cotainer'}`} >
      <div className='background' onClick={closeModal}></div>
      <div className='modal-position modal-container'>
        <span className='span-close' onClick={closeModal}>
          <i className="fa fa-times" aria-hidden="true" ></i>
        </span>
        {children}
      </div>
    </article>
  )
}
