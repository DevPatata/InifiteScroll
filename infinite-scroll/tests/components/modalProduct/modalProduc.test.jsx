import React from 'react'
import { render, screen } from '@testing-library/react';
import { ModalProduct } from "../../../src/components/modalProduct/modalProduct";

describe('Prueba en componente ModalProduct', () => {
  test('Prueba match snaptshot', () => {
    const { container } = render(<ModalProduct />);

    expect(container).toMatchSnapshot(); 
  });

  test('Prueba modal contiene classname "div-cotainer"', () => {
    const isOpen = true;

    render(<ModalProduct isOpen={isOpen}/>);

    expect( screen.getByTestId('modal-test-id').className ).toContain('div-cotainer'); 
  });

  test('Prueba modal no contiene classname "div-cotainer"', () => {
    const isOpen = false;

    render(<ModalProduct isOpen={isOpen}/>);

    expect( screen.getByTestId('modal-test-id').className ).not.toContain('div-cotainer'); 
  });

  test('Prueba modal muestra elemento children', () => {
    const children = 'ModalTest';

    render(<ModalProduct children={children}/>);
    
    expect( screen.getByText('ModalTest') ).toBeTruthy();
  });
})