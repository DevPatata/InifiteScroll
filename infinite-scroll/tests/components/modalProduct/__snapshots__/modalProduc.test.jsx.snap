// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en componente ModalProduct Prueba match snaptshot 1`] = `
<div>
  <article
    class="modal-close false"
    data-testid="modal-test-id"
  >
    <div
      class="background"
    />
    <div
      class="modal-position modal-container"
    >
      <span
        class="span-close"
      >
        <i
          aria-hidden="true"
          class="fa fa-times"
        />
      </span>
    </div>
  </article>
</div>
`;
